internal/kafka/internal/services/kas_fleetshard_operator_addon.go

```diff
@@ -136,7 +144,19 @@ func (o *kasFleetshardOperatorAddon) provisionServiceAccount(clusterId string) (
 	return o.SsoService.RegisterKasFleetshardOperatorServiceAccount(clusterId)
 }
 
-func (o *kasFleetshardOperatorAddon) buildAddonParams(serviceAccount *api.ServiceAccount, clusterId string) []types.Parameter {
+func (o *kasFleetshardOperatorAddon) buildAddonParams(cluster *api.Cluster, serviceAccount *api.ServiceAccount) []types.Parameter {
+
+	var clientId string
+	var clientSecret string
+
+	if cluster.ClientID != "" {
```

Comment:
```suggestion
	if cluster.ClientID != "" && cluster.ClientSecret {
```
We need to also check that the client_id and client_secret are there.